pkg/routeagent_driver/main.go

```diff
@@ -211,3 +213,36 @@ func getTCPMssValue(k8sClientSet *kubernetes.Clientset) int {
 
 	return tcpMssValue
 }
+
+func uninstall(k8sClientSet *kubernetes.Clientset, registry *event.Registry) {
+	if err := registry.StopHandlers(true); err != nil {
+		logger.Warningf("Error stopping handlers: %v", err)
+	}
+
+	if err := annotateNode([]string{}, k8sClientSet); err != nil {
+		logger.Warningf("Error removing %q annotation: %v", constants.CNIInterfaceIP, err)
+	}
+}
+
+func waitForNodeReady(k8sClientSet *kubernetes.Clientset) {
+	// This will wait for ever
+	// In most cases the node will already be ready; otherwise, wait for ever
+	for {
+		localNode, err := node.GetLocalNode(k8sClientSet)
+
+		if err != nil {
+			logger.Error(err, "Error retrieving local node")
+		} else if localNode != nil {
+			for _, condition := range localNode.Status.Conditions {
+				if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+					logger.Info("Node ready")
+					return
+				}
+			}
```

Comment:
import "k8s.io/apimachinery/pkg/api/meta"

```suggestion
                         condition = meta.FindStatusCondition(localNode.Status.Conditions, corev1.NodeReady)
			if condition != nil && condition.Status == corev1.ConditionTrue {
			    logger.Info("Node ready")
			    return
			}
```

Reply:
Nice, thanks!

Reply:
... except that node conditions use their own type, so `FindStatusCondition` can’t be used. There is [a `Node` equivalent](https://pkg.go.dev/k8s.io/kubernetes/pkg/controller/util/node#GetNodeCondition) but it’s not designed for use outside k/k and it can’t be imported without resorting to major `replace` shenanigans.

Reply:
Ah, there is a usable one in `component-helpers`.

Reply:
yeah there's still legacy `Condition` types pre-existing the standard **metav1** `Condition` type.